### PR TITLE
0.8.1

### PR DESCRIPTION
**Pull request by OSVFreeDom**
Updated regex for Clips to support direct channel clip url
Add ClipDownloader to main export
Add Clip as return value of the download function Promise (ClipDownloader), can be useful for apps integration.

### CODE REVIEW

File: src/twitch/clip-downloader.ts [hunk 1]
```diff
@@ -37,8 +37,7 @@ export class ClipDownloader extends EventEmitter {
 
         if (existsSync(appPath(mp4Path))) {
             logger.verbose(`Clip ${clip.title} found at ${appPath(mp4Path)}`);
-
-            return;
+            return Promise.reject(new Error('Clip already exist'));
```
Comment (HugoJF) [inline src/twitch/clip-downloader.ts:40]: You can throw the error directly, I think it's a bit cleaner

File: src/twitch/clip-downloader.ts [hunk 2]
```diff
@@ -59,5 +58,6 @@ export class ClipDownloader extends EventEmitter {
         }
 
         await Promise.all(promises);
+        return Promise.resolve(clip);
```
Comment (HugoJF) [inline src/twitch/clip-downloader.ts:61]: Also return `clip` directly.

File: src/helpers/utils.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {logger}                      from './logger';
 import {Period}                      from '..';
 
 const SPLIT_FACTOR = 2;
-const CLIP_ID_REGEX = /\.tv\/(.+?)(?:\?|$)/;
+const CLIP_ID_REGEX = /[a-zA-Z]+-([a-zA-Z]+([0-9]+[a-zA-Z]+)+)/;
```
Comment (HugoJF) [inline src/helpers/utils.ts:8]: Can you send me a few test cases for this regex? I wanna build a few test cases just to make sure everything keeps working

**Review by HugoJF — CHANGES_REQUESTED** [review]
Minor details

File: src/twitch/clip-downloader.ts (hunk 2 above)
Comment (OSVFreeDom) [reply src/twitch/clip-downloader.ts:61]: Clip is returned line 61

File: src/helpers/utils.ts (hunk 3 above)
Comment (OSVFreeDom) [reply src/helpers/utils.ts:8]: I can't find any clip with pattern twitch.tv/xxx.
All clips are using twitch.tv/channel_name/clip/xxx

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "twitch-tools",
-  "version": "0.8.0",
```
Comment (HugoJF) [inline package.json:3]: You can probably revert this, as I'm currently using a workflow to bump versions and such

File: src/twitch/clip-downloader.ts (hunk 2 above)
Comment (HugoJF) [reply src/twitch/clip-downloader.ts:61]: You don't need to wrap in `Promise.resolve`, you can:
```suggestion
        return clip;
```

File: src/helpers/utils.ts (hunk 3 above)
Comment (HugoJF) [reply src/helpers/utils.ts:8]: This seems to still be working: https://clips.twitch.tv/SassyKawaiiMallardTTours-7vNzQmvsJDGoicMB

**Comment by OSVFreeDom** [comment]
Updates done

**Pull request closed without merging**